api/v1alpha1/openapi.yaml

```diff
@@ -4655,6 +4655,9 @@ components:
       - 'Updating'             # Device
       - 'SpecValid'            # Device (service condition)
       - 'MultipleOwners'       # Device (service condition)
+      - 'DecommissionStarted'  # Device (decommissioning)
+      - 'DecommissionComplete' # Device (decommissioning)
+      - 'DecommissionError'    # Device (decommissioning)
```

Comment:
I believe we are adding one condition `Decommissioning`, and the reason is a golang enum. This allows us to extend the reasons without changing the API spec. This is the pattern used by OCP.

Reply:
Implementation might look like this given above.

```go'
                // default
		updateErr := a.statusManager.UpdateCondition(ctx, v1alpha1.Condition{
			Type:    v1alpha1.DeviceDecommissioning,
			Status:  v1alpha1.ConditionStatusFalse,
		})
		if updateErr != nil {
			a.log.Warnf("Failed setting status: %v", updateErr)
		}
		
		// started
		updateErr := a.statusManager.UpdateCondition(ctx, v1alpha1.Condition{
			Type:    v1alpha1.DeviceDecommissioning,
			Status:  v1alpha1.ConditionStatusTrue,
			Reason:  string(v1alpha1.DecommissionStarted),
			Message: fmt.Sprintf("The device has started the decommission process"),
		})
		if updateErr != nil {
			a.log.Warnf("Failed setting status: %v", updateErr)
		}
		
		// error
		updateErr := a.statusManager.UpdateCondition(ctx, v1alpha1.Condition{
			Type:    v1alpha1.DeviceDecommissioning,
			Status:  v1alpha1.ConditionStatusTrue,
			Reason:  string(v1alpha1.DecommissionError),
			Message: fmt.Sprintf("The device has started the decommissioning process but failed")
		})
		if updateErr != nil {
			a.log.Warnf("Failed setting status: %v", updateErr)
		}
		
	        // success
		updateErr := a.statusManager.UpdateCondition(ctx, v1alpha1.Condition{
			Type:    v1alpha1.DeviceDecommissioning,
			Status:  v1alpha1.ConditionStatusTrue,
			Reason:  string(v1alpha1.DecommissionComplete),
			Message: fmt.Sprintf("The device has been decomissioned")
		})
		if updateErr != nil {
			a.log.Warnf("Failed setting status: %v", updateErr)
		}
		
```

Reply:
Thanks for the example. After looking at this and `internal/agent/device/device.go` I pushed a new version using `Reason`, let me know if I got it right.